### PR TITLE
Added a test for changing an iframe's parentage.

### PR DESCRIPTION
**Pull request by asajeffrey**
Add a test for what happens when an iframe's parent browsing context changes.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] There are tests for these changes

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11677)

### CODE REVIEW

**Comment by ConnorGBrewster** [comment]
Looks good!
@bors-servo r+

-S-awaiting-review 

<details><summary><i>Previously, bors-servo wrote…</i></summary>
> :v: @ConnorGBrewster can now approve this pull request
</details>


---

Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11677#-:-KJlAWisDwV_-8Xm6z1A:1052063960)*

**Comment by asajeffrey** [comment]
IRC chat: http://logs.glob.uno/?c=mozilla%23servo&s=8+Jun+2016&e=8+Jun+2016#c451613

TL;DR: intermittents, gah, and @jgraham asked for the tests to be moved under web-patform-tests.

**Comment by asajeffrey** [comment]
Moved tests to web-platform-tests, rebased and squashed.

**Comment by asajeffrey** [comment]
Grr. Fixed tabs. @bors-servo r=ConnorGBrewster
